### DISCUSSION TITLE
Fastapi and Starlette want Mongodb _id to be iterable (dict type) and it runs into error

### DISCUSSION THREAD

**Discussion by tuxnani** [fastapi/fastapi#6845] [Questions] [answered]
Continuing from [1319](https://github.com/tiangolo/fastapi/issues/1319),

### First check
* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [x] I already read and followed all the tutorial in the docs and didn't find an answer.
* [x] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [x] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [x] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [x] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

### Example

Here's a self-contained, [minimal, reproducible, example](https://stackoverflow.com/help/minimal-reproducible-example) with my use case:
I am trying to return _id field of mongodb documents returned by find().
I get the error :
ValueError: [TypeError("'ObjectId' object is not iterable"), TypeError('vars() argument must have __dict__ attribute')]

```Python
from fastapi import FastAPI

app = FastAPI()


@app.get("/")
def read_root():
    return_string = ""
    cursor = db.collection.find({}).limit(10)
    for doc in cursor:
        return_string += "some text " + str(doc['_id'] + "\n"
    return {"Message": return_string}
```

### Description

* Open the browser and call the endpoint `/`.
* It runs into error.
ValueError: [TypeError("'ObjectId' object is not iterable"), TypeError('vars() argument must have __dict__ attribute')]
* But I expected it to return `{"Message": return strings}`.

### Environment

* OS:  Linux 
* FastAPI Version 0.61.1
* Python version: Python 3.8.5

**Comment by Kludex** [comment]
There's a typo on `doc['_id')]`... But doesn't look like the problem is this one?

**Comment by tuxnani** [comment]
> There's a typo on `doc['_id')]`... But doesn't look like the problem is this one?

Corrected. Thank you.

**Comment by insomnes** [comment]
@tuxnani hello!
Is this `pymongo` or what lib?

**Comment by tuxnani** [comment]
> @tuxnani hello!
> Is this `pymongo` or what lib?

`pymongo` it is.

**Comment by valentin994** [comment]
![image](https://user-images.githubusercontent.com/42819267/108047133-be23be80-7045-11eb-8fc4-054759c4ff92.png)
![image](https://user-images.githubusercontent.com/42819267/108047155-c3810900-7045-11eb-80ea-5e6c9cf8230b.png)
So I tried your code and it works for me 🤔. Or am I missing something?

**Comment by Ayub-Khan** [comment]
following resolved the issue for me.
https://stackoverflow.com/a/18405626

**Comment by tuxnani** [comment, marked as answer]
Typecasting _id or excluding _id from the projection solves the issue.
Thanks to all those who responded!
